### ISSUE TITLE
Wrong hostname in syslog message.

### ISSUE THREAD

**Issue by ket4yii** [rancher/os#1804]
**RancherOS Version: (ros os version)**

0.9.0

**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**

Parallels 6 vm

Rancher writes **rancher** as a hostname(loghost) in syslog instead of **real hostname**

Part of message from syslog:

```
Apr 21 14:03:53 rancher kernel:
Apr 17 16:07:21 rancher sudo:
```

**Comment by SvenDowideit** [comment]
oh wow, yup, we don't bounce the syslog service after a hostname change (and tbh that's hard to detect) but I'll see if i can at least make it work right for hostnames set by the dhcp/cloudinit

**Comment by SvenDowideit** [comment]
so it looks like syslog does use the DHCP based hostname (at least in 1.0.0), but not the one set in the cloud-config file.

```
[   25.072870] time="2017-05-18T05:12:45Z" level=debug msg="Launching Docker /usr/bin/docker docker [daemon --host un
[   26.154969] Bridge firewalling registered                                                                         
[   26.239760] Initializing XFRM netlink socket                                                                      
[   26.257769] IPv6: ADDRCONF(NETDEV_UP): docker0: link is not ready                                                 
[   53.250182] time="2017-05-18T05:13:13Z" level=debug msg="START: [system-docker stop docker] in /home/rancher"
[   76.495387] time="2017-05-18T05:13:36Z" level=debug msg="START: [system-docker start docker] in /home/rancher"
[   76.585712] time="2017-05-18T05:13:36Z" level=debug msg="START: [ros user-docker] in /"
[   77.020944] time="2017-05-18T05:13:37Z" level=debug msg="Warning: user-data hostname (quick2) overrides metadata h
[   77.025616] time="2017-05-18T05:13:37Z" level=info msg="Starting Docker in context: console"
[   77.049142] time="2017-05-18T05:13:37Z" level=debug msg="Warning: user-data hostname (quick2) overrides metadata h
[   77.058657] time="2017-05-18T05:13:37Z" level=info msg="Getting PID for service: console"
[   77.161905] time="2017-05-18T05:13:37Z" level=info msg="console PID 726"          
```

**Comment by SvenDowideit** [comment]
```
[root@quick2 rancher]# cat /etc/hostname 
quick2
[root@quick2 rancher]# system-docker exec -it syslog cat /etc/hostname
ip-172-31-42-202.us-west-2.compute.internal
[root@quick2 rancher]# system-docker exec -it syslog cat /etc/hosts   
127.0.0.1    localhost
::1    localhost ip6-localhost ip6-loopback
fe00::0    ip6-localnet
ff00::0    ip6-mcastprefix
ff02::1    ip6-allnodes
ff02::2    ip6-allrouters

127.0.1.1 quick2
```

**Comment by SvenDowideit** [comment]
this _should_ be fixed in 1.0.2 - I have seen it work, - but it would be good to hear if its working for you :)

**Comment by ket4yii** [comment]
@SvenDowideit Looks good to me. Thanks a lot!
